src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/button_press_request.cc

```diff
@@ -1,20 +1,232 @@
-void ButtonPressRequest::on_event(
-    const application_manager::event_engine::Event& event) {}
+
+void ButtonPressRequest::Execute() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  const std::string button_name =
+      (*message_)[strings::msg_params][message_params::kButtonName].asString();
+  const std::string module_type =
+      (*message_)[strings::msg_params][message_params::kModuleType].asString();
+
+  static ButtonsMap btn_map = buttons_map();
+  mobile_apis::ButtonName::eType button_id =
+      mobile_apis::ButtonName::INVALID_ENUM;
+  if (btn_map.end() != btn_map.find(button_name)) {
+    button_id = btn_map[button_name];
+  }
+
+  const smart_objects::SmartObject* rc_capabilities =
+      application_manager_.hmi_capabilities().rc_capability();
+  const bool button_name_matches_module_type =
+      CheckButtonName(module_type, button_name, rc_capabilities);
+  const bool button_id_exist_in_caps =
+      rc_capabilities &&
+      CheckIfButtonExistInRCCaps(*rc_capabilities, button_id);
+
+  if (button_name_matches_module_type && button_id_exist_in_caps) {
+    SendHMIRequest(hmi_apis::FunctionID::Buttons_ButtonPress,
+                   &(*message_)[strings::msg_params],
+                   true);
+  } else if (!button_name_matches_module_type) {
+    LOG4CXX_WARN(logger_, "Request module type and button name mismatch!");
+    SendResponse(false,
+                 mobile_apis::Result::INVALID_DATA,
+                 "Request module type and button name mismatch!");
+  } else {
+    LOG4CXX_WARN(logger_, "Requested button is not exists in capabilities!");
+    SendResponse(false,
+                 mobile_apis::Result::UNSUPPORTED_RESOURCE,
+                 "Requested button is not exists in capabilities!");
+  }
+}
+
+AcquireResult::eType ButtonPressRequest::AcquireResource(
+    const app_mngr::commands::MessageSharedPtr& message) {
+  const std::string module_type =
+      (*message_)[app_mngr::strings::msg_params][message_params::kModuleType]
+          .asString();
+    app_mngr::ApplicationSharedPtr app =
+      application_manager_.application(CommandRequestImpl::connection_key());
+  return resource_allocation_manager_.AcquireResource(module_type, app->app_id());
+}
+
+bool ButtonPressRequest::IsResourceFree(const std::string& module_type) const {
+  return resource_allocation_manager_.IsResourceFree(module_type);
+}
+
+void ButtonPressRequest::SetResourceState(const std::string& module_type,
+                                          const ResourceState::eType state) {
+  app_mngr::ApplicationSharedPtr app =
+      application_manager_.application(CommandRequestImpl::connection_key());
+  resource_allocation_manager_.SetResourceState(module_type, app->app_id(), state);
+}
+
+void ButtonPressRequest::on_event(const app_mngr::event_engine::Event& event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  RCCommandRequest::on_event(event);
+  DCHECK_OR_RETURN_VOID(
+      (hmi_apis::FunctionID::Buttons_ButtonPress == event.id()));
```

Comment:
@LitvinenkoIra on Driver Consent response also possible. Please avoid dcheks that depends from external input